src/main.ts

```diff
@@ -4,19 +4,49 @@ import minimatch from 'minimatch'
 
 type Format = 'space-delimited' | 'csv' | 'json'
 type FileStatus = 'added' | 'modified' | 'removed' | 'renamed'
+type Quotes = 'none' | 'double' | 'single'
+
+function getQuoter(quotes: Quotes): (files: readonly string[]) => readonly string[] {
+  switch (quotes) {
+    case 'none':
+      return files => files
+    case 'single':
+      return files => files.map(file => `'${file}'`)
+    case 'double':
+      return files => files.map(file => `"${file}"`)
+  }
+}
+
+function getFormatter(format: Format, quotes: Quotes): (files: readonly string[]) => string {
+  const quoter = getQuoter(quotes)
+  switch (format) {
+    case 'space-delimited':
+      return files => quoter(files).join(' ')
+    case 'csv':
+      return files => quoter(files).join(',')
+    case 'json':
+      return files => JSON.stringify(files)
+  }
+}
 
 async function run(): Promise<void> {
   try {
     // Create GitHub client with the API token.
     const client = new GitHub(core.getInput('token', {required: true}))
     const format = core.getInput('format', {required: true}) as Format
     const filter = core.getMultilineInput('filter', {required: true}) || '*'
+    const quotes = (core.getInput('quotes', {}) || 'none') as Quotes
 
     // Ensure that the format parameter is set properly.
     if (format !== 'space-delimited' && format !== 'csv' && format !== 'json') {
       core.setFailed(`Format must be one of 'string-delimited', 'csv', or 'json', got '${format}'.`)
     }
 
+    // Ensure that the quotes parameter is set properly.
+    if (quotes !== 'none' && quotes !== 'single' && quotes !== 'double') {
```

Comment:
we should also check that quotes is `none` for `json`. That case would produce strange results.